### DISCUSSION TITLE
Any ways to access the runner?

### DISCUSSION THREAD

**Discussion by twmht** [open-mmlab/mmengine#889] [General]
if i want to access the current epoch inside the model forwarding, i have to access the runner, but how can i do that?

**Comment by zhouzaida** [comment]
You can access the `epoch` from `MessageHub instance` which is a global object.

```python

class CustomModule:
    def __init__(self):
        self.message_hub = MessageHub.get_current_instance()

    def forward(self, xxx):
        epoch = message_hub.get_info('epoch')
```

References:

- https://github.com/open-mmlab/mmengine/blob/5753cd98e0b791897a2d402b7eb196fb04b4a4eb/mmengine/hooks/runtime_info_hook.py#L77
- https://mmengine.readthedocs.io/en/latest/design/logging.html#messagehub
